src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs

```diff
@@ -0,0 +1,777 @@
+                            {
+                                type = reader.ReadAsInt32(TypePropertyName);
+
+                                if (type == null)
+                                {
+                                    throw new InvalidDataException($"Missing required property '{TypePropertyName}'.");
+                                }
+                            }
+                            else if (reader.TextEquals(InvocationIdPropertyNameBytes))
+                            {
+                                invocationId = reader.ReadAsString(InvocationIdPropertyName);
+                            }
+                            else if (reader.TextEquals(StreamIdsPropertyNameBytes))
+                            {
+                                reader.CheckRead();
+
+                                if (reader.TokenType != JsonTokenType.StartArray)
+                                {
+                                    throw new InvalidDataException($"Expected '{StreamIdsPropertyName}' to be of type {reader.GetTokenString()}.");
+                                }
+
+                                var newStreamIds = new List<string>();
+                                reader.Read();
+                                while (reader.TokenType != JsonTokenType.EndArray)
+                                {
+                                    newStreamIds.Add(reader.GetString());
+                                    reader.Read();
+                                }
+
+                                streamIds = newStreamIds.ToArray();
+                            }
+                            else if (reader.TextEquals(TargetPropertyNameBytes))
+                            {
+                                target = reader.ReadAsString(TargetPropertyName);
+                            }
+                            else if (reader.TextEquals(ErrorPropertyNameBytes))
+                            {
+                                error = reader.ReadAsString(ErrorPropertyName);
+                            }
+                            else if (reader.TextEquals(ResultPropertyNameBytes))
+                            {
+                                hasResult = true;
+
+                                reader.CheckRead();
+
+                                if (string.IsNullOrEmpty(invocationId))
+                                {
+                                    // If we don't have an invocation id then we need to store it as a JsonDocument so we can parse it later
+                                    resultToken = JsonDocument.ParseValue(ref reader);
+                                }
+                                else
+                                {
+                                    // If we have an invocation id already we can parse the end result
+                                    var returnType = binder.GetReturnType(invocationId);
+                                    if (reader.TokenType == JsonTokenType.Null)
```

Comment:
nit: Invert the check to see if `reader.TokenType` isn't null, then you don't need an else clause at all